tests/test_gyrovector_math.py

```diff
@@ -696,3 +696,29 @@ def test_weighted_midpoint_weighted_zero_sum(_k, lincomb):
     assert torch.isfinite(mid).all()
     mid.sum().backward()
     assert torch.isfinite(a.grad).all()
+
+def test_mobius_matvec_vs_log_exp_equivalence():
+    """
+    Test: mobius_matvec = expmap0(W logmap0(x))
+    Now works for high dimensions (e.g., 128), previously only stable for small dims (e.g., 20).
+    """
+    from geoopt.manifolds import Stereographic
+    import torch as th
+
+    dtype = th.float64
+    manifold = Stereographic(k=2.0).to(dtype)
+
+    batch_size = 100
+    dim = 128
+    dim_out = 128
+
+    x = manifold.random_normal(batch_size, dim).to(dtype)
+    weight = th.randn(dim_out, dim, dtype=dtype)
```

Comment:
Let's use these values directly

```suggestion
    x = manifold.random_normal(100, 128).to(dtype)
    weight = th.randn(128, 128, dtype=dtype)
```